osf/models/preprint.py

```diff
@@ -298,14 +299,18 @@ def create(cls, provider, title, creator, description):
         return preprint
 
     @classmethod
-    def create_version(cls, dupliate_from_guid):
+    def create_version(cls, dupliate_from_guid, auth):
         source_preprint = cls.load(dupliate_from_guid.split(cls.GUID_VERSION_DELIMITER)[0])
         if not source_preprint:
             raise NotFound
+        if not source_preprint.has_permission(auth.user, ADMIN):
+            raise PermissionsError('You must have admin permissions to create new version.')
```

Comment:
Yes, we need to check this in the `create_version()`, which is model level check.

However, we also need to configure permission in the API level. We probably need to implement a new permission class with `has_object_permission()` and optional `check_object_permission()` if existing permission class doesn't fit.

Be aware that `has_object_permission()` is automatically called while `check_object_permission()` needs to be manually called.